Includes more components in Intro chap (#25)
centrifuge/developer.centrifuge.io: docs/overview/Introduction.md

## Centrifuge Identity

> A Centrifuge Identity (CentrifugeID) is a unique ID assigned to a participant of the Centrifuge OS in a network. When a new CentrifugeID is generated, a unique public representation of the Identity is created in the public blockchain as a _smart contract_. The data associated with a CentrifugeID is only modifiable by the creator and/or a delegate chosen by the creator.

Reviewer: Why capitalize identity here?

Author: changed

* Ethereuem Accounts: When interacting with a smart contract on Ethereum, an account needs to be linked to the identity to act on its behalf. The linked Ethereum accounts are the accounts that are allowed to interact with DApps utilizing Centrifuge OS.

Centrifuge OS users register their identity on the public Ethereum network and
maintain a set of public keys as part of their identity. These identities and their keys are used to resolve nodes on the P2P layer when one node wants to transmit data to
another Centrifuge user. This means that nodes on the P2P layer find others by

> looking up their public keys from the public identity register based on CentrifugeID.

Reviewer: ```suggestion
looking up their public keys from the public identity register.
```

Author: Done

## Centrifuge Peer-to-Peer Network

> A peer-to-peer (P2P) network consists of two or more systems connected and sharing resources without going through a separate server. Centrifuge consists of a P2P network for decentralized data exchange using smart contracts on Ethereum. The Centrifuge P2P node supports handling of incoming and outgoing data using a service bus principal where plugins and outside systems can subscribe to messages about specific objects.

Reviewer: I wouldn't say "two or more". We have more than two nodes connected at any time. 

These two sentences are a bit repetitive: 
A peer-to-peer (P2P) network consists of two or more systems connected and sharing resources without going through a separate server. Centrifuge consists of a P2P network for decentralized data exchange using smart contracts on Ethereum. 

I wouldn't use the term "service bus". We don't use that anywhere.

## Centrifuge Node

The Centrifuge Node serves an interface to the upstream system in addition to being a part of the peer to peer network that forms the Centrifuge OS ecosystem.<!-- (Need more information here).-->

> ## Centrifuge Sidechain

Reviewer: No chapter on sidechain for now in the developer documentation. This is not part of the current implementation.

Reviewer: We'd also need to make sure we remove the sidechain from the graphic.

The Centrifuge sidechain is a public Ethereum-based blockchain that deploys proof-of-authority consensus algorithm. The collaborators operating on the sidechain are voted in or out based on a token curated registry (TCR) located on the Ethereum mainnet.

> ## Merkel Tree

Reviewer 2: ## Merkle Tree

Author: Done

> In a block, the Merkel tree is a tree in which every lead node is assigned a hash of the data block and every non-leaf node is labelled with a cryptographic hash of the lables of its child nodes. Merkel trees allow you to validate that any element in a large set of values is a part of one Merkel root hash. In the Ethereum blockchain, Merkle trees are created for each block to verify that a specific transaction is part of a mined block.

Reviewer 2: In a block, the _Merkle_ tree is a data structure in which every _leaf_ node is assigned a hash of the data block and every non-leaf node is labelled with a cryptographic hash of the _labels_ of its child nodes. _Merkle_ trees allows to validate that any element in a large set of values is a part of one _Merkle_ root hash. In the Ethereum blockchain, Merkle trees are created for each block to verify that a specific transaction is part of a mined block.

Reviewer: Taking Miguel's changes and improving them a bit:

```suggestion
A Merkle tree is a data structure in which every leaf node is assigned a hash of the data block and every non-leaf node is the hash of the hash of its child nodes. A Merkle tree allows to validate that any element in a large set of values is a part of a set. In the Ethereum blockchain, Merkle trees are created for each block to verify that a specific transaction is part of a mined block. 
```

Author: Done

> Centrifuge OS uses Merkel trees fir proving data validation for documents. In the Centrifuge OS, Merkel trees can be used to share the subset of a document with a third party to prove authenticity of the whole data structure while keeping the rest of the document private.

Reviewer 2: Centrifuge OS uses _Merkle_ trees _for_ proving data validation for documents. In the Centrifuge OS, _Merkle_ trees can be used to share the subset of a document with a third party to prove authenticity of the whole data structure while keeping the rest of the document private.

Author: Done

## Webhooks

> An application using Webhooks will POST a message to an URL when certain events occur. In the Centrifuge OS, webhooks are generally a REST API endpoint that you can set the in the `config.yaml` file. Currently, webhooks for Centrifuge node exist in the Ethereum blockchain and are not maintained by the Centrifuge node.   

Reviewer 2: This sounds confusing:
```Currently, webhooks for Centrifuge node exist in the Ethereum blockchain and are not maintained by the Centrifuge node.```
maybe:
```Currently, webhooks for Centrifuge Node should be developed by the user following our swagger notification API message model.```

Author: Done

## Centrifuge OS Architecture

The core components of the Centrifuge OS are:

* Contracts deployed on Ethereum.

> 

Reviewer: I would not have line breaks between these bullets.

Author: done
